vrs/marten-construction-script.py

```diff
@@ -1,168 +1,180 @@
+ht_union = ht_whole.sample(rand_sample_freq)
+ht_union = ht_union.checkpoint(f'{tmp_path}ht_rand_temp.ht', _read_if_exists=read_if_var, overwrite = overwrite_var)
+
+# Append Table with Indel, Long Reference, Long Variant, and Sex Chromosome variants
+for variant_type in freq_dict:
+    variant_freq = freq_dict[variant_type]
+    desired_count = desired_counts[variant_type]
+    sampling_freq = desired_count / (whole_count * variant_freq)
+    
+    ht_var = ht_whole.filter(ht_whole[variant_type])
+    ht_var = ht_var.sample(sampling_freq)
+    ht_var = ht_var.checkpoint(f'{tmp_path}{variant_type}_temp.ht', _read_if_exists=read_if_var, overwrite=overwrite_var)
+    ht_union = ht_union.union(ht_var)
+
+logger.info(f'Total size of random, indel, long ref, long var, sex chr combined Table as: %i', ht_union.count())
+ht_union = ht_union.checkpoint(f'{tmp_path}union_path_temp.ht', _read_if_exists=read_if_var, overwrite=overwrite_var)
 
 # Construct a table of only multiallelic sites
 
-# for 3 random partitions, only takes the highly multiallelic sites (>5 variants per loci)
+# For 3 random partitions, only takes the highly multiallelic sites (>5 variants per loci)
 ht_filt = ht_whole._filter_partitions(sample(range(ht_whole.n_partitions()), 3))
 group_ht = ht_filt.group_by(ht_filt.locus).aggregate(n=hl.agg.count())
-group_ht = group_ht.filter(group_ht.n > 5)
- 
+group_ht = group_ht.filter(group_ht.n > 5) 
 multiAlleleCount = group_ht.count()
-print(f'There are {multiAlleleCount} very multiallelic SITES in the table above') # 585 sites, plenty of variables
  
-# collects all variants at highly multiallelic loci    
+# Collect all variants at highly multiallelic loci    
+# Do collect() and filter() since it is fairly easy just for one field (locus) , instead of an innter join
 multiList = group_ht.locus.collect(_localize = False)
 ht_multiallelic = ht_filt.filter(multiList.contains(ht_filt.locus))
-# returns 9602 variants on one trial
+logger.info(f'There are %i very multiallelic sites with %i total variants that will be added', multiAlleleCount , ht_multiallelic.count())
 
-# checkpoint for final multiallelic variants
-multiallelic_path = "gs://gnomad-tmp-4day/marten-temp-01-06/multiallelic.ht"
-ht_multiallelic = ht_multiallelic.checkpoint(multiallelic_path, overwrite = True)# , overwrite=overwrite, _read_if_exists=read_if_exists)
-# ^^^ TEMPORARY PATH
+# Checkpoint for final multiallelic variants
+ht_multiallelic = ht_multiallelic.checkpoint(f'{tmp_path}multiallelic_path_temp.ht', _read_if_exists=read_if_var, overwrite = overwrite_var)
 
 # Construct table of variants altered by min-rep
 
-# code in this section provided by Julia Goodrich (best boss this side of the Mississippi)
-vds = hl.vds.read_vds("gs://gnomad/v3.1/raw/gnomad_v3.1.vds")
-ht_minrep = vds.variant_data.rows()
+vds = VersionedVariantDatasetResource(
+    '3.1',
+    {"3.1": VariantDatasetResource("gs://gnomad/v3.1/raw/gnomad_v3.1.vds")},
+)
```

Comment:
you could use one of these here: https://github.com/broadinstitute/gnomad_qc/blob/main/gnomad_qc/v3/resources/basics.py#L154 or https://github.com/broadinstitute/gnomad_qc/blob/main/gnomad_qc/v3/resources/basics.py#L20

example:
```
vds = gnomad_v3_genotypes.vds()
```

Reply:
I tried to do what was shown at line #154 , but with a couple differences, mainly 1) 'gnomad_v3_genotypes_vds' just being named 'vds' and 2) 'CURRENT VERSION' being replaced by just the string '3.1', since previously I couldn't import gnomad_qc (now resolved!). Would you like the code to either use the function at line #20 (since I can import that function now) or look more closely to that at #154 in the linked code?

Reply:
update: got it sorted into one call, so just: 

'ht_minrep = gnomad_v3_genotypes_vds.vds().variant_data.rows()' 

where gnomad_v3_genotypes_vds is imported from 'gnomad_v3.resources.basics'